### PR TITLE
[12.0][IMP] web_responsive: Sticky chatter topbar

### PR DESCRIPTION
**Pull request by Tardo**
Improve chatter topbar as suggested in https://github.com/OCA/web/issues/1177

Sets sticky at top 0 because topbar uses margin, if preserve these margin, when sticky, conversation can be seen in these space because is an a transparent zone.

### CODE REVIEW

**Review by tarteo — APPROVED** [review]

**Review by tarteo — CHANGES_REQUESTED** [review]
Please increase version number

**Comment by Tardo** [comment]
@tarteo updated version number to v1.1.0

**Review by tarteo — APPROVED** [review]
Thanks!

File: web_responsive/static/src/css/web_responsive.scss [hunk 1]
```diff
@@ -420,6 +420,22 @@ html .o_web_client .o_main .o_main_content {
...
+
+                    .o_chatter_topbar,
+                    .o_thread_composer {
+                        position: sticky;
+                    }
+
+                    .o_chatter_topbar {
+                        top: 0;
+                        z-index: 2;
+                        background-color: white;
```
Comment (yajo) [inline web_responsive/static/src/css/web_responsive.scss]: It should be `$o-webclient-background-color` instead. Just to make theming easier. :wink:

**Review by yajo — CHANGES_REQUESTED** [review]
Almost perfect! Just one minor requirement.

Apart from the comment below, I've been testing and it seems a little bit confusing that the attachments panel isn't also sticky. This is not a requirement, but I think that if it is easy to fix, it'd be good to fix it too. What do you think?
peek 14-02-2019 08-31

**Comment by Tardo** [comment]
@Yajo idk how to stack sticky elements dynamically only with CSS... the problem it's that the composer and attachments can be show at the same time (or not).

**Review by yajo — APPROVED** [review]
Hmm I have been playing around and it seems like the only sane way is to put `.o_chatter_topbar`, `.o_thread_composer` and `.o_mail_chatter_attachments` inside the same div, which should be the sticky one:

peek 15-02-2019 10-53

(Look at the new div highlighted in blue in the dom panel, and its css properties highlighted below in green).

However, after thinking more about it, maybe a record with too many attachments would lose its ability to scroll over them, so better leave as it is.

**Comment by pedrobaeza** [comment]
Thanks to all.

**Comment by Tardo** [comment]
@Yajo Oh! xD good solution :P and can be useful to preserve margin top. For the scroll issue something like this can work:
```
.o_attachments_previews {
  overflow: auto;
  max-height: $o-mail-attachment-image-size * 2.5 * 3.0;
}
```
I try it?

**Comment by pedrobaeza** [comment]
Let's try to see how it's seen!

**Comment by yajo** [comment]
Another solution would be to make the right panel scroll up if the user clicks on clippy.

**Pull request merged**